ex1/functions.py

```diff
@@ -91,25 +147,81 @@ def f():
 # dictionary with the position of the argument as the key (starting at 0) and
 # the argument as the value.
 
+def args_to_dict(a,b,c):
+    """merges a,b, and c into a dict with their argument position as the key
+    """
+    return {0:a,1:b,2:c}
+    
+def args_to_dict_general(*args):
+    """merges a,b, and c into a dict with their argument position as the key
+    """
+    i=0
+    res = {}
+    for arg in args:
+        res[i]=arg
+        i+=1
+        
+    return res
+
 # BONUS: Write a function named args_to_dict_general that does the same for any
 # number of arguments
 
 # Define a function named lists_to_dict that takes two lists of equal lenght
 # named keys and values and builds a dictionary out of them.
 
+def lists_to_dict(keys,values):
+    """creates a list with keys and values taken from the input argument lists
+    """
+    return dict(zip(keys,values))
+
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
 
+def search_list(a,b):
+    """searches for elements b in a and returns indices/values in a dictionary
+    """
+    dic = {}    
+    for val in b:
+        try:
+            dic[a.index(val)]=val
+        except:
+            pass
+    return dic
+
 # Define a function named dict_to_string that takes a dictionary and a
 # separator string. The function should only take elements out of the
 # dictionary whose value is a string and then return a single string containing
 # the strings stored in the dictionary seperated by the separator string.
 # Return an empty string if there are no strings in the dictionary.
 
+def dict_to_string(d,s):
+    """Collect strings out of the dictionary d and returns them speartated by s
+    """
+    res = ''
+    for el in d:
+        if isinstance(d[el],str): 
+            res += d[el] + s
+        
+    return res[0:-1]
```

Comment:
Making a list of the strings and using the `join` method is cleaner than using `+` and then removing the last separator manually. See https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20/files#diff-7ec7b04e6cd2fbdbc70b7b64bf731ae8R392 for an example